src/snowflake/snowpark/__init__.py

```diff
@@ -34,8 +34,6 @@
     "WhenNotMatchedClause",
     "QueryRecord",
     "QueryHistory",
-    "AsyncJob",
```

Comment:
AsyncJob is still needed, but _AsyncDataType is not needed. Any module prefixed with _ is considered for internal usage only.

Reply:
@sfc-gh-yuwang ^ this one

Reply:
got it